File: escenario4_docker-http-python/benchmark.py
# benchmark.py
import requests
import time
import statistics

URL = "http://127.0.0.1:8080"
ITERACIONES = 100

latencias = []

for i in range(ITERACIONES):
    start = time.time()
    resp = requests.get(URL)
    elapsed = (time.time() - start) * 1000  # convertir a ms
    latencias.append(elapsed)

    # opcional: imprimir cada resultado
    print(f"Iteración {i+1}: {elapsed:.3f} ms")

# calcular métricas
promedio = statistics.mean(latencias)
minimo = min(latencias)
maximo = max(latencias)

print("\n📊 Resultados del benchmark")
print(f"Solicitudes: {ITERACIONES}")
print(f"Latencia promedio: {promedio:.3f} ms")
print(f"Latencia mínima: {minimo:.3f} ms")
print(f"Latencia máxima: {maximo:.3f} ms")

File: escenario2_docker-http-go/benchmark.go
// benchmark.go
package main

import (
    "fmt"
    "net/http"
    "time"
    "io/ioutil"
)

func main() {
    url := "http://127.0.0.1:8080"
    iteraciones := 100
    latencias := make([]float64, 0, iteraciones)

    for i := 0; i < iteraciones; i++ {
        start := time.Now()
        resp, err := http.Get(url)
        if err != nil {
            panic(err)
        }
        _, _ = ioutil.ReadAll(resp.Body)
        resp.Body.Close()
        elapsed := time.Since(start).Seconds() * 1000 // ms
        latencias = append(latencias, elapsed)
        fmt.Printf("Iteración %d: %.3f ms\n", i+1, elapsed)
    }

    // calcular métricas
    var suma, min, max float64
    min, max = latencias[0], latencias[0]
    for _, v := range latencias {
        suma += v
        if v < min {
            min = v
        }
        if v > max {
            max = v
        }
    }
    promedio := suma / float64(iteraciones)

    fmt.Println("\n📊 Resultados del benchmark")
    fmt.Printf("Solicitudes: %d\n", iteraciones)
    fmt.Printf("Latencia promedio: %.3f ms\n", promedio)
    fmt.Printf("Latencia mínima: %.3f ms\n", min)
    fmt.Printf("Latencia máxima: %.3f ms\n", max)
}
